chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler_test.go

```diff
@@ -0,0 +1,767 @@
+			},
+			wantErr: true,
+		},
+		{
+			name: "failure: mongo error while retrieving the experiment run details",
+			given: func() {
+				ctx = context.WithValue(ctx, authorization.AuthKey, username)
+				findResult := []interface{}{bson.D{
+					{Key: "experiment_id", Value: experimentId},
+					{Key: "experiment_runs", Value: []*dbChaosExperimentRun.ChaosExperimentRun{
+						{ExperimentRunID: experimentRunID},
+					}},
+				}}
+				singleResult := mongo.NewSingleResultFromDocument(findResult[0], nil, nil)
+				mongodbMockOperator.On("Get", mock.Anything, mongodb.ChaosExperimentCollection, mock.Anything).Return(singleResult, nil).Once()
+
+				chaosExperimentService.On("ProcessExperimentUpdate", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+				mongodbMockOperator.On("Get", mock.Anything, mongodb.ChaosExperimentRunsCollection, mock.Anything).Return(singleResult, errors.New("error retrieving the experiment runs")).Once()
+			},
+			wantErr: true,
+		},
+		{
+			name: "failure: unable to delete experiment-runs",
+			given: func() {
+				ctx = context.WithValue(ctx, authorization.AuthKey, username)
+				findResult := []interface{}{bson.D{
+					{Key: "experiment_id", Value: experimentId},
+					{Key: "experiment_runs", Value: []*dbChaosExperimentRun.ChaosExperimentRun{
+						{ExperimentRunID: experimentRunID},
+					}},
+				}}
+				singleResult := mongo.NewSingleResultFromDocument(findResult[0], nil, nil)
+				mongodbMockOperator.On("Get", mock.Anything, mongodb.ChaosExperimentCollection, mock.Anything).Return(singleResult, nil).Once()
+
+				chaosExperimentService.On("ProcessExperimentUpdate", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+				mongodbMockOperator.On("Get", mock.Anything, mongodb.ChaosExperimentRunsCollection, mock.Anything).Return(singleResult, nil).Once()
+				chaosExperimentService.On("ProcessExperimentDelete", mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil).Once()
+
+				mongodbMockOperator.On("Get", mock.Anything, mongodb.ChaosExperimentRunsCollection, mock.Anything).Return(singleResult, nil).Once()
+
+				chaosExperimentRunService.On("ProcessExperimentRunDelete", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(errors.New("")).Once()
+			},
+			wantErr: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.given()
+			_, err := chaosExperimentHandler.DeleteChaosExperiment(ctx, projectId, experimentId, &experimentRunID, store)
+			if (err != nil) != tc.wantErr {
+				t.Errorf("ChaosExperimentHandler.DeleteChaosExperiment() error = %v, wantErr %v", err, tc.wantErr)
+				return
+			}
+		})
+	}
+}
+
+func TestChaosExperimentHandler_UpdateChaosExperiment(t *testing.T) {
```

Comment:
This testcase is failed because new operations added since last commit.

`UpdateChaosExperiment` use this function below

https://github.com/litmuschaos/litmus/blob/794676f6b75e08835334fd5d675e0b0d096d5f9d/chaoscenter/graphql/server/pkg/chaos_experiment/handler/handler.go#L248-L251

and this operator use a global variable

https://github.com/litmuschaos/litmus/blob/794676f6b75e08835334fd5d675e0b0d096d5f9d/chaoscenter/graphql/server/pkg/database/mongodb/chaos_experiment/operations.go#L130

Can you change this one to local variable and update testcase? 

Other than that, LGTM 🚀

Reply:
Made the changes in latest commit!